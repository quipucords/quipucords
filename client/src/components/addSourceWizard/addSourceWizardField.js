import React from 'react';
import PropTypes from 'prop-types';
import { Form, Grid } from 'patternfly-react';
import helpers from '../../common/helpers';

const AddSourceWizardField = ({ children, colLabel, colField, id, label, error, errorMessage, ...props }) => {
  const setId = id || helpers.generateId();

  return (
    <Form.FormGroup controlId={setId} validationState={error ? 'error' : null} {...props}>
      <Grid.Col componentClass={Form.ControlLabel} sm={colLabel}>
        {label}
      </Grid.Col>
      <Grid.Col sm={colField}>
        {children}
        {error && <Form.HelpBlock>{errorMessage}</Form.HelpBlock>}
      </Grid.Col>
    </Form.FormGroup>
  );
};

AddSourceWizardField.propTypes = {
  children: PropTypes.node.isRequired,
  colLabel: PropTypes.number,
  colField: PropTypes.number,
  id: PropTypes.string,
  label: PropTypes.node,
  error: PropTypes.string,
  errorMessage: PropTypes.string
};

AddSourceWizardField.defaultProps = {
  colLabel: 3,
  colField: 9,
  id: null,
  label: null,
  error: null,
  errorMessage: null
};

export { AddSourceWizardField as default, AddSourceWizardField };
