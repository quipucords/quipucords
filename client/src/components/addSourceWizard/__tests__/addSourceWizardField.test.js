import React from 'react';
import { mount } from 'enzyme';
import { AddSourceWizardField } from '../addSourceWizardField';

describe('AddSourceWizardField Component', () => {
  it('should render', () => {
    const props = { id: 'test' };

    const component = mount(
      <AddSourceWizardField {...props}>
        <input id="test" type="text" value="" readOnly />
      </AddSourceWizardField>
    );

    expect(component.render()).toMatchSnapshot('render');
    expect(
      component
        .find('label')
        .at(0)
        .render()
    ).toMatchSnapshot('label');
  });
});
