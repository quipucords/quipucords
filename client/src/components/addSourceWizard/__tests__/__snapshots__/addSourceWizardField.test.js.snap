// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label 1`] = `
<label
  class="col-sm-3 control-label"
  for="test"
/>
`;

exports[`render 1`] = `
<div
  class="form-group"
>
  <label
    class="col-sm-3 control-label"
    for="test"
  />
  <div
    class="col-sm-9"
  >
    <input
      id="test"
      readonly=""
      type="text"
      value=""
    />
  </div>
</div>
`;
